File: src/main/java/musiccube/entities/Country.java
package musiccube.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;


@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
@Table(name = "country")
public class Country {

    @Id
    @Column(name = "id")
    @GeneratedValue
    private int id;

    @Column(unique = true)
    private String countryName;

    @Column(unique = true)
    private String code;

    public Country() {
    }

    public Country(String countryName, String code) {
        this.countryName = countryName;
        this.code = code;
    }

    public int getId() {
        return id;
    }

    public String getCountryName() {
        return countryName;
    }

    public void setCountryName(String countryName) {
        this.countryName = countryName;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }
}

File: Frontend/src/app/Class/Country.ts
export class Country {
  id: number;
  countryName: string;
  code: string;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.countryName = (obj && obj.countryName) || '';
    this.code = (obj && obj.code) || '';
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getCountryName(): string {
    return this.countryName;
  }

  getCode(): string {
    return this.code;
  }

  // Setters

  setCountryName(countryName: string) {
    this.countryName = countryName;
  }

  setCode(code: string) {
    this.code = code;
  }
  
}
